test/steinertree/steiner_tree.jl

```diff
@@ -0,0 +1,51 @@
+@testset "Steiner Tree" begin
+    
+    function sum_weight(g::AbstractGraph{<:Integer}, distmx::AbstractMatrix{<:Integer} = weights(g))
+        sum_wt = zero(eltype(g))
+        for e in edges(g)
+            sum_wt += distmx[src(e), dst(e)]
+        end
+        return sum_wt
+    end
+
+
+    g3 = StarGraph(5) 
+    for g in testgraphs(g3)
+        g_st = @inferred(steiner_tree(g, [2, 5]))
+        @test ne(g_st) == 2 # [Edge(2, 1), Edge(1, 5)]
+
+        g_copy = SimpleGraph(g)
+        LightGraphs.filter_non_term_leaves!(g_copy, [2, 5])
+        @test ne(g_copy) == 2 # [Edge(2, 1), Edge(1, 5)]
+    end
+
+    g4 = PathGraph(11) 
+    for g in testgraphs(g4)
+        g_st = @inferred(steiner_tree(g, [4, 8]))
+        @test ne(g_st) == 4
+
+        g_copy = SimpleGraph(g)
+        LightGraphs.filter_non_term_leaves!(g_copy, [4, 8])
+        @test ne(g_copy) == 4
+    end
+
+
+    d = [0   2   3   4   5
+         2   0   60   80  100
+         3   60   0  120  150
+         4   80  120  0  200
+         5  100  150  200  0]
+
+    g5 = CompleteGraph(5) 
+    for g in testgraphs(g5)
+        g_st = @inferred(steiner_tree(g, [2, 4, 5], d))
+        @test sum_weight(g_st, d) == 11 #2+4+5
+    end
+
+    g6 = Grid([5, 5])
+    for g in testgraphs(g6)
+        g_st = @inferred(steiner_tree(g, [3, 11, 15, 23]))
+        @test sum_weight(g_st) == 8 
+    end
+
```

Comment:
maybe add also a more complicated tests (maybe randomized? )  where you verify that `sum_weight` is within the theoretical approximation bounds.

Reply:
I will need to know the optimal solution to find the theoretical bound.
This could be done for fixed weights but difficult to automate for randomised weights. (I will need to implement an exponential time algo to find the minimum)

Reply:
Done with fixed weights

Reply:
Ah yeah, I forgot, that we don't have a correct solution.